### PR TITLE
 docs: update `Overview.md` to fix several issues

### PR DESCRIPTION
**Pull request by HollowMan6**
Fix singular noun after plural quantifier:
snapshots -> snapshot

Also addressed the suggestions by @efeg :

> * `workload` -> `load` (i.e. the term `workload` is no longer the preferred way to refer to the `load`. For consistency, can we adopt `load` in this document?).
> * `snapshot` -> `window` (i.e. the term `snapshot` was used to be the preferred way to refer to a `window`. To make it more confusing, the documentation also uses the term "time window" to refer to the period from the start to the end of a `window`. Can we update those to improve the clarity?)
> * there are also other occurrences of the same issue (e.g. `each of sampling tasks...`, `Each sampling tasks...`)
> * `Kafka Cruise Control` -> `Cruise Control`

### CODE REVIEW

File: docs/wiki/Overview.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Each metric fetcher will be assigned a few number of partitions in the cluster t
...
 
-When queried, the metric sampler aggregator will return all the aggregated workload snapshot back to the LoadMonitor. Users can define a minimum number of metric samples required in each workload snapshot to ensure the workload snapshot has enough metric samples. For example, if user specifies snapshot window to be 1 hour and the sampling interval is 5 minutes, ideally each snapshots should have 12 metric samples. User can define that a snapshot window with less than 10 metric samples is invalid and should not be included into the cluster workload model.
+When queried, the metric sampler aggregator will return all the aggregated workload snapshot back to the LoadMonitor. Users can define a minimum number of metric samples required in each workload snapshot to ensure the workload snapshot has enough metric samples. For example, if user specifies snapshot window to be 1 hour and the sampling interval is 5 minutes, ideally each snapshot should have 12 metric samples. User can define that a snapshot window with less than 10 metric samples is invalid and should not be included into the cluster workload model.
```
Comment (efeg) [inline docs/wiki/Overview.md:32]: Here are some other issues that we can fix in this document:
* `workload` -> `load` (i.e. the term `workload` is no longer the preferred way to refer to the `load`. For consistency, can we adopt `load` in this document?).
* `snapshot` -> `window` (i.e. the term `snapshot` was used to be the preferred way to refer to a `window`. To make it more confusing, the documentation also uses the term "time window" to refer to the period from the start to the end of a `window`. Can we update those to improve the clarity?)
* there are also other occurrences of the same issue (e.g. `each of sampling tasks...`, `Each sampling tasks...`)
* `Kafka Cruise Control` -> `Cruise Control`

**Review by efeg — COMMENTED** [review]
@HollowMan6 Thanks for the PR!
Now that you fixed one issue, I added a comment to point out a few more :)
Maybe this PR can help us fix those, too.

Once the comments are addressed, can we also update the title of this PR to reflect the overall changes?

File: docs/wiki/Overview.md (hunk 1 above)
Comment (HollowMan6) [reply docs/wiki/Overview.md]: Done! Thanks for your advice.

**Review by efeg — APPROVED** [review]
Thanks for the updates - LGTM.

**Pull request merged**